libp2p/protocols/pubsub/pubsub.nim

```diff
@@ -43,9 +48,40 @@ type
   MsgIdProvider* =
     proc(m: Message): string {.noSideEffect, raises: [Defect], nimcall, gcsafe.}
 
+  TopicParams* = object
+    topicWeight*: float
+
+    # p1
+    timeInMeshWeight*: float
+    timeinMeshQuantum*: Duration
+    timeInMeshCap*: float
+
+    # p2
+    firstMessageDeliveriesWeight*: float
+    firstMessageDeliveriesDecay*: float
+    firstMessageDeliveriesCap*: float
+
+    # p3
+    meshMessageDeliveriesWeight*: float
+    meshMessageDeliveriesDecay*: float
+    meshMessageDeliveriesThreshold*: float
+    meshMessageDeliveriesCap*: float
+    meshMessageDeliveriesActivation*: Duration
+    meshMessageDeliveriesWindow*: Duration
+
+    # p3b
+    meshFailurePenaltyWeight*: float
+    meshFailurePenaltyDecay*: float
+
+    # p4
+    invalidMessageDeliveriesWeight*: float
+    invalidMessageDeliveriesDecay*: float
+
   Topic* = object
```

Comment:
This isn't the right place for it, we should move this into gossip 1.1 and simply extend the `Topic` object.

Reply:
Wip, Yep already moved. 
I honestly was hoping to avoid double table lookups but we are not using topics table in gossip anyway! So indeed was pointless 😀

Reply:
Afraid that some data will be in `pubsubpeer` btw, shame on nim to not have protocols and interface. I'll try to avoid when possible but to the limits of rationality in terms of lookups and tables complexity.